controllers/metal3.io/host_state_machine.go

```diff
@@ -226,6 +230,34 @@ func (hsm *hostStateMachine) checkInitiateDelete() bool {
 	return true
 }
 
+func (hsm *hostStateMachine) checkDeregisteredHost(info *reconcileInfo) (result actionResult) {
+	// If the deregistered annotation is set we remove any host from the
+	// provisioner and take no further action
+	// Note this doesn't change the current state, only the OperationalStatus
+	annotations := hsm.Host.GetAnnotations()
+	if annotations != nil {
+		if _, ok := annotations[metal3v1alpha1.DeregisteredAnnotation]; ok {
+			if info.host.OperationalStatus() != metal3v1alpha1.OperationalStatusDeregistered {
+				_, err := hsm.Provisioner.Delete()
+				if err != nil {
+					return actionError{errors.Wrap(err, "failed to deregister deregistered host")}
+				}
+				info.host.SetOperationalStatus(metal3v1alpha1.OperationalStatusDeregistered)
+				info.log.Info("host is deregistered, removed from provisioner")
+				return actionComplete{}
+			}
+			return deregisterComplete{}
```

Comment:
Elsewhere we've adopted a policy that we should never just stop reconciling a Host altogether. e.g. in actionUnmanaged we return actionContinue with unmanagedRetryDelay (which is 10 minutes). It would seem appropriate to do the same here.

Reply:
@zaneb I was a little confused about that - what's the reason for the periodic retry even for resources in a static/done state?  Won't any change to the resource trigger reconciliation without the periodic retry?

Reply:
Yes, in theory it's not required. I think we just decided as a general design principle that we didn't want to ever just start completely ignoring a resource. I don't have a memory of what specific risks we thought that might help reduce. Maybe @dhellmann does?

Reply:
Switched to unmanagedRetryDelay in the latest version